harness/determined/exec/tensorboard.py

```diff
@@ -26,121 +41,137 @@ def set_s3_region() -> None:
-    by downstream processes to determine args passed in.
-    :return: version in the form of (major, minor) tuple
-    """
-
-    major, minor, _ = version.split(".")
-
-    return major, minor
-
-
-def get_tensorboard_args(args: List[str]) -> List[str]:
-    """
-    Builds tensorboard startup args from args passed in from tensorboard-entrypoint.sh
     Args are added and deprecated at the mercy of tensorboard; all of the below are necessary to
     support versions 1.14, 2.4, and 2.5
 
-    - If multiple directories are specified and the tensorboard version is > 1,
-    use legacy logdir_spec behavior
-
     - Tensorboard 2+ no longer exposes all ports. Must pass in "--bind_all" to expose localhost
-
     - Tensorboard 2.5.0 introduces an experimental feature (default load_fast=true)
     which prevents multiple plugins from loading correctly.
     """
     task_id = os.environ["DET_TASK_ID"]
     port = os.environ["TENSORBOARD_PORT"]
 
-    version = args.pop(0)
-
-    # logdir is the second argument passed in from tensorboard_manager.go. If multiple directories
-    # are specified and the tensorboard version is > 1, use legacy logdir_spec behavior. NOTE:
-    # legacy logdir_spec behavior is not supported by many tensorboard plugins
-    logdir = args.pop(0)
-
     tensorboard_args = [
         "tensorboard",
         f"--port={port}",
         f"--path_prefix=/proxy/{task_id}",
-        *args,
+        *add_args,
     ]
 
-    major, minor = get_tensorboard_version(version)
-
-    if major == "2":
+    # Version dependant args
+    if version.parse(tb_version) >= version.parse("2"):
         tensorboard_args.append("--bind_all")
-        if minor >= "5":
-            tensorboard_args.append("--load_fast=false")
-        if len(logdir.split(",")) > 1:
-            tensorboard_args.append(f"--logdir_spec={logdir}")
-            return tensorboard_args
+    if version.parse(tb_version) >= version.parse("2.5"):
+        tensorboard_args.append("--load_fast=false")
 
-    tensorboard_args.append(f"--logdir={logdir}")
-    return tensorboard_args
+    tensorboard_args.append(f"--logdir={tfevents_dir}")
 
+    return tensorboard_args
 
-def main(args: List[str]) -> int:
-    set_s3_region()
 
+def get_tensorboard_url() -> str:
     task_id = os.environ["DET_TASK_ID"]
     port = os.environ["TENSORBOARD_PORT"]
     tensorboard_addr = f"http://localhost:{port}/proxy/{task_id}"
-    url = f"{tensorboard_addr}/data/plugin/scalars/tags"
-    tensorboard_args = get_tensorboard_args(args)
+    return f"{tensorboard_addr}/data/plugin/scalars/tags"
+
+
+def check_for_metrics() -> bool:
+    tensorboard_url = get_tensorboard_url()
+    tags = {}
+    try:
+        # Attempt to retrieve metrics from tensorboard.
+        res = requests.get(tensorboard_url)
+        res.raise_for_status()
+        logger.debug(
+            f"requests.get({tensorboard_url}) -> " "Response.content: {!r}".format(res.content)
```

Comment:
Why not combine these strings?  Or can you put a `+` in between the strings?

Reply:
This was an annoying MyPy formatting thing. It doesn't like it when I printf a string and bytes, even though I intent to print bytes. Are you cool with that?

Reply:
Done. This was a weird MyPy issue along with auto-formatting strangeness.